pkg/aspect/lint/bep.go

```diff
@@ -231,35 +231,41 @@ func (runner *LintBEPHandler) bepEventHandler(event *buildeventstream.BuildEvent
 			for _, fileSetId := range outputGroup.FileSets {
 				if fileSet := runner.namedSets[fileSetId.Id]; fileSet != nil {
 					runner.namedSets[fileSetId.Id] = nil
-					result := runner.resultsByLabel[label]
-					if result == nil {
-						result = &ResultForLabel{label: label}
-						runner.resultsByLabel[label] = result
+
+					// go through the fileSet and create a result for each mnemonic
+					for _, file := range fileSet.GetFiles() {
+						if mnemonic := parseLinterMnemonicFromFilename(file.Name); mnemonic != "" {
+							if _, ok := runner.resultsByLabelByMnemonic[label+mnemonic]; !ok {
+								// create a new result for this label and mnemonic
+								result := &ResultForLabelAndMnemonic{label: label, mnemonic: mnemonic}
+								runner.resultsByLabelByMnemonic[label+mnemonic] = result
+							}
```

Comment:
Important to not create a new result struct if one has already been constructed.